CoreScriptsRoot/CoreScripts/Topbar.lua

```diff
@@ -957,17 +979,42 @@ local function Create3DSettingsIcon(topBarInstance, panel, menubar)
 
 		BackgroundTransparency = 1,
 
-		Image = "rbxasset://textures/ui/Menu/hamburger3D.png"
+		Image = "rbxasset://textures/ui/VR/hamburger.png"
 	}
 
+	local menuState = false
+	GuiService.MenuOpened:connect(function() menuState = true end)
+	GuiService.MenuClosed:connect(function() menuState = false end)
+
+	function menuItem:OnSetActive(active)
+		MenuModule:SetVisibility(active)
+		if not active then
```

Comment:
Hmm, the point was to make sure CloseAllExcept would stall until the animation completes. Perhaps it would be better to do GuiService.MenuClosed:wait(), now that I think of it. I'll change CloseAllExcept to CloseAllExceptAsync.